### DISCUSSION TITLE
Thoughts on recent up down arrows on first and last line.

### DISCUSSION THREAD

**Discussion by MShark67** [TurboPack/SynEdit#121] [Ideas]
I think this is definitely one of those "what you're used to seems correct" issues and I'm not used to this behavior (but I also don't see it as a major issue btw.)  It is a bit of an issue if you use the eoKeepCaretX feature as just pressing up on the top line or down on the bottom line loses your X position.  Would it make any sense to make this behavior reliant on eoKeepCaretX?  It seems like it's a good use for that option as I don't think it would have any adverse side effects.

```
          if (CaretY = 1) and not (eoKeepCaretX in Options) then
            DoHomeKey(Command = ecSelUp)
```
-Mark

**Comment by pyscripter** [comment]
It is the default behavior in Word, VS Code and Visual Studio (not sure about other major text editors), so it can't be too surprising to users.

But your suggestion deserves some thinking.

**Comment by MShark67** [comment]
I'll get used to it either way.  I do notice that Word only does it for shift-up/down (which I had never noticed.)  My favorite PHP/HTML editor appears to do it but only at the bottom of the file and not at the top.
